src/zenml/integrations/gcp/step_operators/vertex_step_operator.py

```diff
@@ -250,7 +250,14 @@ def launch(
                             "boot_disk_size_gb": settings.boot_disk_size_gb,
                         },
                     }
-                ]
+                ],
+                "service_account": self.config.service_account,
+                "network": self.config.network,
+                "reserved_ip_ranges": (
+                    self.config.reserved_ip_ranges
+                    if self.config.reserved_ip_ranges
+                    else []
```

Comment:
for my understanding, why do we take in a string for this in the flavor, but it's a list here when defining?

Reply:
hmmm yep I agree there's a smelly thing overthere..  Let me check (need to test deeper)

Reply:
(I think GH Copilot misled me...)

Reply:
@strickvl  am I right that the [parse_name_and_extra_arguments](https://github.com/zenml-io/zenml/blob/cdd14528d674e0cafe00173755e556ea00c6343c/src/zenml/cli/utils.py#L787) utility function that handles the specific args of a (step operator) flavor doesn't handle list type parameters (i.e. same behavior as the `multiple=True` option in `@click.option()` decorator)?

What is the standard way of handling such _list parameters_ in ZenML CLI ?

Reply:
I think till now we've managed to avoid this situation :) AFAICS, `parse_name_and_extra_arguments` would need to be updated to handle the case of a list of inputs, but let me check with a colleague and get back to you.

Reply:
So as far as I understand it, I think the `parse_name_and_extra_arguments` currently doesn't handle that so would need to be updated to take care of a string that represented a list of arguments.